File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

def dfs(v):
    visited_dfs[v]=1
    print(v,end=' ')
    
    for i in range(1, n + 1):
        #방문한적 없고 다른곳에 연결되어 있으면
        if not visited_dfs[i] and graph[v][i] == 1:
            dfs(i)
    
def bfs(v):
    queue = deque([v])
    visited_bfs[v]= 1
    while queue:
        v = queue.popleft()
        print(v,end=' ')
        # 아직 방문하지 않은 인접한 원소들을 큐에 삽입
        for i in range(1, n + 1):
            #방문기록이 없고, 인덱스에 값이 있다면(연결되어있다면)
            if not visited_bfs[i] and graph[v][i] == 1:
                queue.append(i)
                visited_bfs[i]=1
                
                
n,m,v=map(int,input().split())

# 그래프 생성
graph = [[0]*(n+1) for _ in range(n+1)]
for i in range(m):
    a,b=map(int,input().split())
    graph[a][b]=graph[b][a]=1

visited_dfs = [0]*(n+1)
visited_bfs = [0]*(n+1)

dfs(v)
print()
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const input = require("fs")
  .readFileSync("./dev/stdin")
  .toString()
  .trim()
  .split("\n");

const [n, _, v] = input.shift().split(" ").map(Number);
const edges = input.map((v) => v.split(" ").map(Number));

let graph = Array.from({ length: n + 1 }, () =>
  Array.from({ length: n + 1 }).fill(0)
);

edges.forEach(([a, b]) => {
  graph[a][b] = 1;
  graph[b][a] = 1;
});

let visited_dfs = Array.from({ length: n + 1 }).fill(0);
let visited_bfs = Array.from({ length: n + 1 }).fill(0);

dfs(v);
console.log("");
bfs(v);

function dfs(v) {
  visited_dfs[v] = 1;
  process.stdout.write(v + " ");
  for (let i = 1; i <= n; i++) {
    if (graph[v][i] === 1 && visited_dfs[i] === 0) {
      dfs(i);
    }
  }
}
function bfs(v) {
  let queue = [v];
  visited_bfs[v] = 1;
  while (queue.length > 0) {
    v = queue.shift();
    process.stdout.write(v + " ");
    for (let i = 1; i <= n; i++) {
      if (graph[v][i] === 1 && visited_bfs[i] === 0) {
        queue.push(i);
        visited_bfs[i] = 1;
      }
    }
  }
}
